data_import/admin.py

```diff
@@ -62,5 +63,25 @@ def save_model(self, request, obj, form, change):
         super().save_model(request, obj, form, change)
 
 
+class AdminStandardizedFile(admin.ModelAdmin):
+    form = UploadForm
+
+    def save_model(self, request, obj, form, change):
+        upload = Upload.objects.create()
+
+        uploaded_file = form.cleaned_data['standardized_file']
+
+        s_file_meta = {
+            'standardized_file': uploaded_file,
+            'upload': upload,
+            'reporting_year': form.cleaned_data['reporting_year'],
+        }
+
+        s_file = StandardizedFile.objects.create(**s_file_meta)
+
```

Comment:
You already have access to a `StandardizedFile` instance through `obj`. You'll want to operate on that instance, instead of creating a new one. Look to `save_model` on the source file admin class for an example.

https://github.com/datamade/bga-payroll/blob/8f00a32746f28d35e278043e9cfdc949afa3f6af/data_import/admin.py#L52-L62

Reply:
This makes sense, but on inspecting the `obj`, `obj` doesn't have an `id`. I'm not sure how to make sure that `obj` has an id before getting to this method. I could write logic to set one, but that doesn't feel like a good solution. Any thoughts?

Reply:
`obj` has not been saved yet. An `id` will be generated when you save it, [either directly or by calling `super().save_model()`](https://docs.djangoproject.com/en/3.0/ref/contrib/admin/#django.contrib.admin.ModelAdmin.save_model).